node/pkg/telemetry/telemetry.go

```diff
@@ -38,7 +40,23 @@ var logLevelSeverity = map[zapcore.Level]logging.Severity{
 	zapcore.FatalLevel:  logging.Emergency,
 }
 
-func (enc *encoder) EncodeEntry(entry zapcore.Entry, fields []zapcore.Field) (*buffer.Buffer, error) {
+func (enc *guardianTelemetryEncoder) EncodeEntry(entry zapcore.Entry, fields []zapcore.Field) (*buffer.Buffer, error) {
+
+	// if skipPrivateLogs==true, then private logs don't go to telemetry
+	if enc.skipPrivateLogs {
+		for _, f := range fields {
+			if f.Type == zapcore.BoolType {
+				if f.Key == "_privateLogEntry" {
+					if f.Integer == 1 {
+						return enc.Encoder.EncodeEntry(entry, fields)
```

Comment:
doesn't this do the opposite i.e. encode logs when _privateLogEntry == true?

Reply:
by returning early here, it short-circuits the publishing to telemetry by going to the underlying encoder directly. I added a corresponding code comment for clarity.